src/grid.h

```diff
@@ -668,7 +668,7 @@ struct Grid {
     }
 
     void MergeWithParent(Grid *p, Selection &s, Document *doc) {
-        cell->grid = nullptr;
+        cell->grid.release();

```

Comment:
This is still not so great code, because you remove the pointer from the unique_ptr, and then "throw it away". We know there is an owner elsewhere, but that is not obvious from this code. So at the very least there should be a comment indicating who is the new owner and why. Ideally, it be even better if the code was structured such that `release` was used at the very moment the new owner takes the pointer, but only if that doesn't otherwise complicate the code.

Reply:
You can merge this commit and think about the above feedback for a separate commit, just to stop the crashing.

Reply:
Ok I will think about your feedback. The issue in my opinion is that the pasted grid is cloned into the grid owned by the cell and that is it, or the same happens but the grid is directly dissolved because it is merged. So there is a fifty fifty chance that the owner keeps the grid or loses it directly after gaining it.

Reply:
Ok, but in that case maybe some more comments in the code locations where this happens would be good.